src/log.rs

```diff
@@ -112,17 +162,26 @@ mod tests {
 
         // filter created
         transport.add_response(json!("0xf0"));
+        // get past logs
+        transport.add_response(json!([generate_log("awesome0")]));
         // get logs filter
-        transport.add_response(json!([generate_log("awesome")]));
-
-        let log = LogStream::new(web3, Default::default(), Duration::from_secs(0))
-            .next()
-            .immediate()
-            .expect("log stream did not produce any logs")
-            .expect("failed to get log from log stream");
-
-        assert_eq!(log.log_type.as_deref(), Some("awesome"));
+        transport.add_response(json!([generate_log("awesome1")]));
+
+        let mut stream = LogStream::new(web3, Default::default(), Duration::from_secs(0));
+        let mut logs = Vec::new();
+        for _ in 0..2 {
+            let log = stream
```

Comment:
You might be able to use some stream combinators to make this feel more like iterators.